tests/test_generate_framework_master_csv.py

```diff
@@ -0,0 +1,156 @@
+    assert mock_data_client.get_framework.called_once_with('test_framework_slug')
+
+    csv_builder.service_status_labels = OrderedDict([('key1', 'test_lot_prefix_1'), ('key2', 'test_lot_prefix_2')])
+    csv_builder.static_fieldnames = ('test_static_fieldname_1', 'test_static_fieldname_2')
+
+    expected_data = tuple([
+        'test_static_fieldname_1',
+        'test_static_fieldname_2',
+        'test_lot_prefix_1_test_lot_slug_1',
+        'test_lot_prefix_2_test_lot_slug_1',
+        'test_lot_prefix_1_test_lot_slug_2',
+        'test_lot_prefix_2_test_lot_slug_2'
+    ])
+    assert csv_builder.get_fieldnames() == expected_data
+
+
+@mock.patch(
+    'dmscripts.generate_framework_master_csv.GenerateMasterCSV.get_fieldnames',
+    return_value=['test_field_1', 'test_field_2'],
+)
+def test_write_csv(fieldname_mock, mock_data_client):
+    """Test writing csv."""
+    csv_builder = GenerateMasterCSV(client=mock_data_client, target_framework_slug='test_framework_slug')
+    assert fieldname_mock.called_once_with('test_framework_slug')
+    csv_builder.output = [
+        {'test_field_1': 'foo', 'test_field_2': 'bar'},
+        {'test_field_1': 'baz', 'test_field_2': 'Ąćĉ'}
+    ]
+    f = StringIO()
+    csv_builder.write_csv(outfile=f)
+
+    expected_data = "test_field_1,test_field_2\nfoo,bar\nbaz,Ąćĉ\n"
+
+    assert f.getvalue() == expected_data
+
+
+def test_populate_output_suppliers(mock_data_client):
+    """Test supplier info, no lot info."""
+    with open('tests/fixtures/test_supplier_frameworks_response.json') as supplier_frameworks_response:
+        mock_data_client.find_framework_suppliers.return_value = json.loads(supplier_frameworks_response.read())
+    csv_builder = GenerateMasterCSV(client=mock_data_client, target_framework_slug='test_framework_slug')
+    f = StringIO()
+    csv_builder.populate_output()
+    csv_builder.write_csv(outfile=f)
+
+    with open('tests/fixtures/test_populate_output_suppliers_expected_result.csv') as expected_file:
+        assert f.getvalue() == expected_file.read()
+
+
+def test_one_supplier_one_lot(mock_data_client):
+    """Test a single client in a single lot."""
+    mock_data_client.get_framework.return_value = {
+        'frameworks': {'lots': [{'slug': 'saas'}]}
+    }
+    mock_data_client.find_framework_suppliers.return_value = {
+        'supplierFrameworks': [
+            {'supplierId': 123, 'supplierName': 'Bens cool supplier', 'extraneous_field': 'foo', 'declaration': ''}
+        ]
+    }
+    mock_data_client.find_draft_services_iter.return_value = iter([{
+        'lot': 'saas', 'lotSlug': 'saas', 'status': 'submitted', 'extraneous_field': 'foo'
+    }])
+
+    csv_builder = GenerateMasterCSV(client=mock_data_client, target_framework_slug='test_framework_slug')
+    f = StringIO()
+    csv_builder.populate_output()
+    csv_builder.write_csv(outfile=f)
+
+    with open('tests/fixtures/test_one_supplier_one_lot_result.csv') as expected_file:
```

Comment:
Changed them to be relative to file with:

```FIXTURES_DIR = os.path.join(os.path.dirname(__file__), 'fixtures')```
Easier to refactor directory structure and `py.test` can be run from anywhere. 
👍